### PR TITLE
Add /maxNativeAmount endpoint

### PR DESCRIPTION
**Pull request by lok52**
Changes:
* Add `/maxNativeAmount` endpoint
* Change `RELAYER_MAX_NATIVE_AMOUNT_FAUCET` parameter name to `RELAYER_MAX_NATIVE_AMOUNT`
* Change native amount validation algorithm

### CODE REVIEW

File: zp-relayer/validation/tx/validateTx.ts [hunk 1]
```diff
@@ -65,10 +65,9 @@ export function checkTxSpecificFields(txType: TxType, tokenAmount: BN, energyAmo
   return null
 }
 
-export function checkNativeAmount(nativeAmount: BN | null) {
+export function checkNativeAmount(nativeAmount: BN | null, withdrawalAmount: BN) {
   logger.debug(`Native amount: ${nativeAmount}`)
-  // Check native amount (relayer faucet)
-  if (nativeAmount && nativeAmount > config.maxFaucet) {
+  if (nativeAmount && (nativeAmount > config.maxNativeAmount || nativeAmount > withdrawalAmount)) {
```
Comment (k1rill-fedoseev) [inline zp-relayer/validation/tx/validateTx.ts:70]: How does it work? Should it be `X.gt(Y)` instead of `X > Y`?

Comment (EvgenKor) [reply zp-relayer/validation/tx/validateTx.ts:70]: I confirm that issue. In the currently deployed relayer the `nativeAmount` compared with `config.maxNativeAmount` unpredictably incorrect

Comment (lok52) [reply zp-relayer/validation/tx/validateTx.ts]: Fixed

**Review by EvgenKor — APPROVED** [review]
Looks good to me. Need to try it out with the associated client library PR https://github.com/zkBob/zkbob-client-js/pull/126

**Review by k1rill-fedoseev — APPROVED** [review]

**Review by akolotov — COMMENTED** [review]
Should the new endpoint be reflected in README.md?

File: zp-relayer/validation/tx/validateTx.ts [hunk 2]
```diff
@@ -65,10 +65,9 @@ export function checkTxSpecificFields(txType: TxType, tokenAmount: BN, energyAmo
   return null
 }
 
-export function checkNativeAmount(nativeAmount: BN | null) {
+export function checkNativeAmount(nativeAmount: BN | null, withdrawalAmount: BN) {
   logger.debug(`Native amount: ${nativeAmount}`)
-  // Check native amount (relayer faucet)
-  if (nativeAmount && nativeAmount > config.maxFaucet) {
+  if (nativeAmount && (nativeAmount.gt(config.maxNativeAmount) || nativeAmount.gt(withdrawalAmount))) {
```
[the remark this answers is not in the document]
Comment (lok52) [reply zp-relayer/validation/tx/validateTx.ts]: Fixed and refactored validation of delta fields. I hope, the validation of token and energy amounts is more expressive, now

**Pull request merged**